src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -95,26 +99,120 @@ private void setSubmissionData(Map<String, Object> data, String overseasEntityId
+                if (Objects.nonNull(beneficialOwner.getTrustIds())) {
+                    for (String trustId : beneficialOwner.getTrustIds()) {
+                        List<TrustDataDto> trustData = new ArrayList<>();
+
+                        if (!Objects.nonNull(submissionDto.getTrusts())) {
+                            throw new ServiceException("No trusts exist for this filing but a trust id is provided for BO Individual "
+                                    + beneficialOwner.getFirstName() + " " + beneficialOwner.getLastName());
+                        }
+                        for (TrustDataDto trust : submissionDto.getTrusts()) {
+                            if (trust.getTrustId().equals(trustId)) {
+                                trustData.add(trust);
+                            }
+                        }
+
+                        // If there is more than one trust with the same ID throw an error
+                        if (trustData.size() > 1) {
+                            throw new ServiceException("There is more than one trust with the ID: " + trustId);
+                        }
+                        // If there are is no trust with that ID throw an error
+                        if (trustData.isEmpty()) {
+                            throw new ServiceException("There are no trusts for the ID: " + trustId);
+                        }
+
+                        trustsDataForBO.add(trustData.get(0));
+                    }
+                }
+
+                String trustData = "";
+                if (!trustsDataForBO.isEmpty()) {
+                    // Convert trust data to JSON string if it exists on transaction else it's to an empty string
+                    ObjectMapper mapper = JsonMapper.builder().findAndAddModules().build();
+                    try {
+                        trustData = mapper.writeValueAsString(trustsDataForBO);
+                    } catch (JsonProcessingException e) {
+                        throw new ServiceException("Error converting trust data to JSON " + e.getMessage(), e);
+                    }
+                }
+
+                beneficialOwner.setTrustData(trustData);
+                beneficialOwnersIndividualSubmissionData.add(beneficialOwner);
             }
         }
-        data.put(TRUST_DATA, trustData);
 
-        ApiLogger.debug("Submission data has been set on filing", logMap);
+        return beneficialOwnersIndividualSubmissionData;
+    }
+
+    private List<BeneficialOwnerCorporateDto> getBeneficialOwnersCorporateSubmissionData(OverseasEntitySubmissionDto submissionDto) throws ServiceException {
+        List<BeneficialOwnerCorporateDto> beneficialOwnersCorporateSubmissionData = new ArrayList<>();
+
+        if (Objects.nonNull(submissionDto.getBeneficialOwnersCorporate())) {
+            for (BeneficialOwnerCorporateDto beneficialOwner : submissionDto.getBeneficialOwnersCorporate()) {
```

Comment:
Parts of the body of this method are more or less an exact match of the above method. If you're familiar with lambda functions and passing them in as method parameters you could refactor to have a common method. This might become clearer after doing the method split suggest above.